### ISSUE TITLE
Null provider is not download

### ISSUE THREAD

**Issue by quentin9696** [hashicorp/terraform#25310]
### Terraform Version

```
Terraform v0.13.0-beta2
```

### Terraform Configuration Files

See here : https://github.com/danieldreier/terraform-issue-reproductions/pull/6

### Debug Output

`terraform init`

```
2020/06/19 09:36:51 [WARN] Log levels other than TRACE are currently unreliable, and are supported only for backward compatibility.
  Use TF_LOG=TRACE to see Terraform's internal logs.
  ----
2020/06/19 09:36:51 [INFO] Terraform version: 0.13.0 beta2 
2020/06/19 09:36:51 [INFO] Go runtime version: go1.14.2
2020/06/19 09:36:51 [INFO] CLI args: []string{"/usr/local/bin/terraform_013", "init"}
2020/06/19 09:36:51 [DEBUG] Attempting to open CLI config file: /home/quentin/.terraformrc
2020/06/19 09:36:51 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2020/06/19 09:36:51 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2020/06/19 09:36:51 [DEBUG] ignoring non-existing provider search directory /home/quentin/.terraform.d/plugins
2020/06/19 09:36:51 [DEBUG] ignoring non-existing provider search directory /home/quentin/.local/share/terraform/plugins
2020/06/19 09:36:51 [DEBUG] ignoring non-existing provider search directory /home/quentin/.local/share/flatpak/exports/share/terraform/plugins
2020/06/19 09:36:51 [DEBUG] ignoring non-existing provider search directory /var/lib/flatpak/exports/share/terraform/plugins
2020/06/19 09:36:51 [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2020/06/19 09:36:51 [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2020/06/19 09:36:51 [INFO] CLI command args: []string{"init"}
2020/06/19 09:36:51 [WARN] Log levels other than TRACE are currently unreliable, and are supported only for backward compatibility.
  Use TF_LOG=TRACE to see Terraform's internal logs.
  ----
Initializing modules...
2020/06/19 09:36:51 [WARN] Log levels other than TRACE are currently unreliable, and are supported only for backward compatibility.
  Use TF_LOG=TRACE to see Terraform's internal logs.
  ----
2020/06/19 09:36:51 [DEBUG] Module installer: begin foo
2020/06/19 09:36:51 [DEBUG] Module installer: foo installed at ../..
2020/06/19 09:36:51 [DEBUG] Module installer: begin foo.bar
- foo in ../..
2020/06/19 09:36:51 [DEBUG] Module installer: foo.bar installed at ../../modules/bar
- foo.bar in ../../modules/bar

Initializing the backend...
2020/06/19 09:36:51 [DEBUG] New state was assigned lineage "a4349015-3afd-d708-8c14-f8999f51baad"
2020/06/19 09:36:51 [DEBUG] checking for provisioner in "."
2020/06/19 09:36:51 [DEBUG] checking for provisioner in "/usr/local/bin"
2020/06/19 09:36:51 [INFO] Failed to read plugin lock file .terraform/plugins/linux_amd64/lock.json: open .terraform/plugins/linux_amd64/lock.json: no such file or directory

2020/06/19 09:36:51 [WARN] Failed to scan provider cache directory .terraform/plugins: cannot search .terraform/plugins: lstat .terraform/plugins: no such file or directory
2020/06/19 09:36:51 [DEBUG] Service discovery for registry.terraform.io at https://registry.terraform.io/.well-known/terraform.json
Initializing provider plugins...
- Finding hashicorp/template versions matching "~> 2.1"...
2020/06/19 09:36:51 [WARN] Log levels other than TRACE are currently unreliable, and are supported only for backward compatibility.
  Use TF_LOG=TRACE to see Terraform's internal logs.
  ----
2020/06/19 09:36:51 [DEBUG] GET https://registry.terraform.io/v1/providers/hashicorp/template/versions
2020/06/19 09:36:52 [WARN] Log levels other than TRACE are currently unreliable, and are supported only for backward compatibility.
  Use TF_LOG=TRACE to see Terraform's internal logs.
  ----
2020/06/19 09:36:52 [DEBUG] GET https://registry.terraform.io/v1/providers/hashicorp/template/2.1.2/download/linux/amd64
2020/06/19 09:36:52 [DEBUG] GET https://releases.hashicorp.com/terraform-provider-template/2.1.2/terraform-provider-template_2.1.2_SHA256SUMS
2020/06/19 09:36:52 [DEBUG] GET https://releases.hashicorp.com/terraform-provider-template/2.1.2/terraform-provider-template_2.1.2_SHA256SUMS.sig
- Installing hashicorp/template v2.1.2...
2020/06/19 09:36:56 [DEBUG] Provider signed by 51852D87348FFC4C HashiCorp Security <security@hashicorp.com>
- Installed hashicorp/template v2.1.2 (signed by HashiCorp)

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

And `terraform validate`

```
2020/06/19 09:38:18 [WARN] Log levels other than TRACE are currently unreliable, and are supported only for backward compatibility.
  Use TF_LOG=TRACE to see Terraform's internal logs.
  ----
2020/06/19 09:38:18 [INFO] Terraform version: 0.13.0 beta2 
2020/06/19 09:38:18 [INFO] Go runtime version: go1.14.2
2020/06/19 09:38:18 [INFO] CLI args: []string{"/usr/local/bin/terraform_013", "validate"}
2020/06/19 09:38:18 [DEBUG] Attempting to open CLI config file: /home/quentin/.terraformrc
2020/06/19 09:38:18 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /home/quentin/.terraform.d/plugins
2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /home/quentin/.local/share/terraform/plugins
2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /home/quentin/.local/share/flatpak/exports/share/terraform/plugins
2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /var/lib/flatpak/exports/share/terraform/plugins
2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2020/06/19 09:38:18 [INFO] CLI command args: []string{"validate"}
2020/06/19 09:38:18 [WARN] Log levels other than TRACE are currently unreliable, and are supported only for backward compatibility.
  Use TF_LOG=TRACE to see Terraform's internal logs.
  ----
2020/06/19 09:38:18 [DEBUG] checking for provisioner in "."
2020/06/19 09:38:18 [DEBUG] checking for provisioner in "/usr/local/bin"
2020/06/19 09:38:18 [INFO] Failed to read plugin lock file .terraform/plugins/linux_amd64/lock.json: open .terraform/plugins/linux_amd64/lock.json: no such file or directory
2020-06-19T09:38:18.712-0400 [INFO]  plugin: configuring client automatic mTLS
2020-06-19T09:38:18.739-0400 [DEBUG] plugin: starting plugin: path=.terraform/plugins/registry.terraform.io/hashicorp/template/2.1.2/linux_amd64/terraform-provider-template_v2.1.2_x4 args=[.terraform/plugins/registry.terraform.io/hashicorp/template/2.1.2/linux_amd64/terraform-provider-template_v2.1.2_x4]
2020-06-19T09:38:18.740-0400 [DEBUG] plugin: plugin started: path=.terraform/plugins/registry.terraform.io/hashicorp/template/2.1.2/linux_amd64/terraform-provider-template_v2.1.2_x4 pid=51968
2020-06-19T09:38:18.740-0400 [DEBUG] plugin: waiting for RPC address: path=.terraform/plugins/registry.terraform.io/hashicorp/template/2.1.2/linux_amd64/terraform-provider-template_v2.1.2_x4
2020-06-19T09:38:18.745-0400 [INFO]  plugin.terraform-provider-template_v2.1.2_x4: configuring server automatic mTLS: timestamp=2020-06-19T09:38:18.745-0400
2020-06-19T09:38:18.773-0400 [DEBUG] plugin.terraform-provider-template_v2.1.2_x4: plugin address: address=/tmp/plugin766504208 network=unix timestamp=2020-06-19T09:38:18.773-0400
2020-06-19T09:38:18.773-0400 [DEBUG] plugin: using plugin: version=5
2020-06-19T09:38:18.829-0400 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unimplemented desc = unknown service plugin.GRPCStdio"
2020-06-19T09:38:18.830-0400 [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/hashicorp/template/2.1.2/linux_amd64/terraform-provider-template_v2.1.2_x4 pid=51968
2020-06-19T09:38:18.830-0400 [DEBUG] plugin: plugin exited
2020-06-19T09:38:18.830-0400 [DEBUG] plugin: starting plugin: path=/usr/local/bin/terraform_013 args=[/usr/local/bin/terraform_013, internal-plugin, provisioner, local-exec]
2020-06-19T09:38:18.830-0400 [DEBUG] plugin: plugin started: path=/usr/local/bin/terraform_013 pid=51983
2020-06-19T09:38:18.830-0400 [DEBUG] plugin: waiting for RPC address: path=/usr/local/bin/terraform_013
2020-06-19T09:38:18.842-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [WARN] Log levels other than TRACE are currently unreliable, and are supported only for backward compatibility.
2020-06-19T09:38:18.842-0400 [DEBUG] plugin.terraform_013:   Use TF_LOG=TRACE to see Terraform's internal logs.
2020-06-19T09:38:18.842-0400 [DEBUG] plugin.terraform_013:   ----
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [INFO] Terraform version: 0.13.0 beta2 
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [INFO] Go runtime version: go1.14.2
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [INFO] CLI args: []string{"/usr/local/bin/terraform_013", "internal-plugin", "provisioner", "local-exec"}
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [DEBUG] Attempting to open CLI config file: /home/quentin/.terraformrc
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /home/quentin/.terraform.d/plugins
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /home/quentin/.local/share/terraform/plugins
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /home/quentin/.local/share/flatpak/exports/share/terraform/plugins
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /var/lib/flatpak/exports/share/terraform/plugins
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2020-06-19T09:38:18.855-0400 [DEBUG] plugin.terraform_013: 2020/06/19 09:38:18 [INFO] CLI command args: []string{"internal-plugin", "provisioner", "local-exec"}
2020-06-19T09:38:18.856-0400 [DEBUG] plugin.terraform_013: local-exec-provisioner (internal) 2020/06/19 09:38:18 [INFO] Starting provisioner plugin local-exec
2020-06-19T09:38:18.856-0400 [DEBUG] plugin.terraform_013: plugin address: address=/tmp/plugin092188818 network=unix timestamp=2020-06-19T09:38:18.856-0400
2020-06-19T09:38:18.856-0400 [DEBUG] plugin: using plugin: version=5
2020-06-19T09:38:18.857-0400 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2020-06-19T09:38:18.859-0400 [DEBUG] plugin: plugin process exited: path=/usr/local/bin/terraform_013 pid=51983
2020-06-19T09:38:18.859-0400 [DEBUG] plugin: plugin exited

Error: Could not load plugin


Plugin reinitialization required. Please run "terraform init".

Plugins are external binaries that Terraform uses to access and manipulate
resources. The configuration provided requires plugins which can't be located,
don't satisfy the version constraints, or are otherwise incompatible.

Terraform automatically discovers provider requirements from your
configuration, including providers used in child modules. To see the
requirements and constraints, run "terraform providers".

Failed to instantiate provider "registry.terraform.io/hashicorp/null" to
obtain schema: unknown provider "registry.terraform.io/hashicorp/null"
```

### Expected Behavior

The null provider should be downloaded. 

### Actual Behavior

Terraform null provider is missing

### Steps to Reproduce

1. `terraform init examples/standard`
2. `terraform apply examples/standard`

### Additional Context

`terraform providers`

```
Providers required by configuration:
.
└── module.foo
    ├── provider[registry.terraform.io/hashicorp/template] ~> 2.1
    └── module.bar
        └── provider[registry.terraform.io/hashicorp/null] ~> 2.1
```

A quick workaround is to defined the null provider in the `required_providers` like this : 

examples/standard/main.tf
```hcl
...
terraform {
  required_version = ">= 0.13"

  required_providers {
    null = {
      source = "hashicorp/null"
    }
  }
}
```
Also, this didn't happened if I tried to run the terraform from the root directory.

**Comment by debovema** [comment]
Related to #25282 & #25312.

**Comment by alisdair** [comment]
Thanks for the detailed reproduction case! This bug has been fixed in #25334, which will be included in the next 0.13.0 prerelease.
